src/Infrastructure/Services/TokenProviderService.cs

```diff
@@ -20,9 +21,30 @@ public TokenProviderService(IOptions<TokenProviderServiceSettings> settings)
 
         public async Task<string> AcquireTokenAsync(string resource)
         {
-            var assertionCert = CertificateUtility.GetByThumbprint(_settings.CertificateThumbprint);
-            var clientAssertion = new ClientAssertionCertificate(_settings.ClientId, assertionCert);
+            if (string.IsNullOrEmpty(_settings.CertificateThumbprint) && string.IsNullOrEmpty(_settings.ClientSecret))
```

Comment:
Isn't that ||, not && here- string.IsNullOrEmpty(_settings.CertificateThumbprint) && string.IsNullOrEmpty(_settings.ClientSecret)

Reply:
I believe we want it the way it is.  This checks to see if both settings are empty and the next check checks if both are non-empty.  In either case, something is wrong as we just want them to fill in one or the other setting.  Does that sound right to you too?